api/proto/sign/v1alpha1/sign_service.proto

```diff
@@ -0,0 +1,88 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package sign.v1alpha1;
+
+import "core/v1alpha1/agent.proto";
+import "google/protobuf/empty.proto";
+
+service SignService {
+  // Sign agents using keyless OIDC based provider
+  rpc SignOIDC(SignOIDCRequest) returns (SignOIDCResponse);
+  // Sign agents using PEM-encoded private key encrypted with an optional passphrase
+  rpc SignWithKey(SignWithKeyRequest) returns (SignWithKeyResponse);
+
+  // Verify signed agents using keyless OIDC based provider
+  rpc VerifyOIDC(VerifyOIDCRequest) returns (google.protobuf.Empty);
+
+  // Verify signed agents using PEM-encoded formatted PEM public key encrypted
+  rpc VerifyWithKey(VerifyWithKeyRequest) returns (google.protobuf.Empty);
```

Comment:
both commands can be done as part of two interfaces `Sign` and `Verify`. the messages for each can leverage proto `oneOf` to indicate what kind of signing and verification process is requested -- either OIDC or Key. IMO, this does not have to flow via separate methods, but rather as part of the requests